quipucords/fingerprinter/tests_fingerprint_task.py

```diff
@@ -1228,12 +1251,23 @@ def test_scan_all_facts_with_null_value_in_process_satellite_scan(self):
             metadata_dict,
         )
         expected_fingerprints = {
-            fingerprint_name: False
+            fingerprint_name: None
             for fingerprint_name in EXPECTED_FINGERPRINT_MAP_SATELLITE
         }
-        expected_fingerprints[PRODUCTS_KEY] = mock.ANY
-        expected_fingerprints[ENTITLEMENTS_KEY] = []
+        expected_fingerprints["is_redhat"] = False
         expected_fingerprints["infrastructure_type"] = SystemFingerprint.UNKNOWN
+        expected_fingerprints[ENTITLEMENTS_KEY] = []
+
+        copy_products_list = deepcopy(PRODUCTS)
+        for product in copy_products_list:
+            product["metadata"] = {
+                "server_id": self.server_id,
+                "source_name": "source3",
+                "source_type": "satellite",
+                "raw_fact_key": None,
+            }
+        expected_fingerprints[PRODUCTS_KEY] = copy_products_list
```

Comment:
I'll keep my original implementation on this one because the solution above doesn't quite work. The dict should look like this {product dict , "metadata" : {                 server_id=self.server_id,
                source_name="source3",
                source_type=Source.SATELLITE_SOURCE_TYPE,
                raw_fact_key=None,
}}

Reply:
Ok! in this case the original has better readability